### PR TITLE
S_ISDIR is already defined and std::ifstream<wchar_t> is not supporte…

### PR DESCRIPTION
**Pull request by testdrive-profiling-master**
1. S_ISDIR is already defined in MinGW
2. std::ifstream<wchar_t> is not supported in MinGW, only ifstream<char>
   -> wrapping with "std::filesystem::path(...)"

### CODE REVIEW

File: src/common/util/src/file_util.cpp [hunk 1]
```diff
@@ -33,7 +33,9 @@
 /// @brief Windows-specific 'mkdir' wrapper
 #    define makedir(dir) _mkdir(dir)
 // Copied from linux libc sys/stat.h:
-#    define S_ISDIR(m) (((m)&S_IFMT) == S_IFDIR)
+#    if !defined(__MINGW32__) && !defined(__MINGW32__)
```
Comment (rkazants) [inline src/common/util/src/file_util.cpp:36]:
```suggestion
#    if !defined(__MINGW32__) && !defined(__MINGW64__)
```

Comment (rkazants) [reply src/common/util/src/file_util.cpp:36]: ?

Comment (testdrive-profiling-master) [reply src/common/util/src/file_util.cpp]: thanks

**Comment by rkazants** [comment]
build_jenkins

**Review by rkazants — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
@testdrive-profiling-master please, fix clang format code style issues

**Comment by ilya-lavrenov** [comment]
build_jenkins

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**